src/parser/TurtleParser.cpp

```diff
@@ -371,86 +372,105 @@ bool TurtleParser<T>::rdfLiteral() {
-      if (type == XSD_INT_TYPE || type == XSD_INTEGER_TYPE ||
-          type == XSD_NON_POSITIVE_INTEGER_TYPE ||
-          type == XSD_NEGATIVE_INTEGER_TYPE || type == XSD_LONG_TYPE ||
-          type == XSD_SHORT_TYPE || type == XSD_BYTE_TYPE ||
-          type == XSD_NON_NEGATIVE_INTEGER_TYPE ||
-          type == XSD_UNSIGNED_LONG_TYPE || type == XSD_UNSIGNED_INT_TYPE ||
-          type == XSD_UNSIGNED_SHORT_TYPE ||
-          type == XSD_POSITIVE_INTEGER_TYPE) {
-        parseIntegerConstant(strippedLiteral);
-      } else if (type == XSD_BOOLEAN_TYPE) {
-        if (strippedLiteral == "true") {
-          lastParseResult_ = true;
-        } else if (strippedLiteral == "false") {
-          lastParseResult_ = false;
-        } else {
-          LOG(DEBUG)
-              << strippedLiteral
-              << " could not be parsed as a boolean object of type " << type
-              << ". It is treated as a plain string literal without datatype "
-                 "instead"
-              << std::endl;
-          lastParseResult_ = std::move(previous);
-        }
-      } else if (type == XSD_DECIMAL_TYPE || type == XSD_DOUBLE_TYPE ||
-                 type == XSD_FLOAT_TYPE) {
-        parseDoubleConstant(strippedLiteral);
-      } else if (type == XSD_DATETIME_TYPE) {
-        lastParseResult_ = DateOrLargeYear::parseXsdDatetime(strippedLiteral);
-      } else if (type == XSD_DATE_TYPE) {
-        lastParseResult_ = DateOrLargeYear::parseXsdDate(strippedLiteral);
-      } else if (type == XSD_GYEARMONTH_TYPE) {
-        lastParseResult_ = DateOrLargeYear::parseGYearMonth(strippedLiteral);
-      } else if (type == XSD_GYEAR_TYPE) {
-        lastParseResult_ = DateOrLargeYear::parseGYear(strippedLiteral);
+    literalAndDatatypeToTripleComponentImpl(
+        asStringViewUnsafe(previous.getContent()), lastParseResult_.getIri(),
+        this);
+  }
+
+  // It is okay to neither have a langtag nor an XSD datatype.
+  return true;
+}
+
+// ______________________________________________________________________
+template <class T>
+TripleComponent TurtleParser<T>::literalAndDatatypeToTripleComponentImpl(
+    std::string_view normalizedLiteralContent,
+    const TripleComponent::Iri& typeIri, TurtleParser<T>* parser) {
+  auto literal =
+      TripleComponent::Literal::literalWithoutQuotes(normalizedLiteralContent);
+  std::string_view type = asStringViewUnsafe(typeIri.getContent());
+
+  try {
+    // TODO<joka921> clean this up by moving the check for the types to a
+    // separate module.
+    if (type == XSD_INT_TYPE || type == XSD_INTEGER_TYPE ||
+        type == XSD_NON_POSITIVE_INTEGER_TYPE ||
+        type == XSD_NEGATIVE_INTEGER_TYPE || type == XSD_LONG_TYPE ||
+        type == XSD_SHORT_TYPE || type == XSD_BYTE_TYPE ||
+        type == XSD_NON_NEGATIVE_INTEGER_TYPE ||
+        type == XSD_UNSIGNED_LONG_TYPE || type == XSD_UNSIGNED_INT_TYPE ||
+        type == XSD_UNSIGNED_SHORT_TYPE || type == XSD_POSITIVE_INTEGER_TYPE) {
```

Comment:
Maybe just add a member to the 
`TurtleParser`class that contains those types all (as a static `vector` or `array` etc.) then you can simply use
`if (ad_utility::contains(integerDatatypes(), type) {....}` 
And something similar for the three floating point types.

This will
1. Make Codecov happier
2. Get rid of the TODO here.